### DISCUSSION TITLE
Images are not loading in storefront and dashboard

### DISCUSSION THREAD

**Discussion by sycts88** [saleor/saleor#5715] [Q&A]
### What I'm trying to achieve
I am trying to change the images in saleor dashboard 2.10.  

### Steps to reproduce the problem
1.  I have deployed the saleor dashboard 2.10, saleor storefront and master to heroku using heroku button
2.   when i opened saleor dashboard, product images are not shown, saleor storefront images also not shown
3. I have kept the images in S3 AWS

### What I expected to happen
Images for products should shows in dashboard and saleor storefront

### Screenshots
![image](https://user-images.githubusercontent.com/59194717/83145089-dbff6b00-a126-11ea-8022-5a5aeb08e620.png)

**System information**
Operating system:
Windows 10

**Comment by sycts88** [comment]
anyone can help on the above issue about images are not loading in storefront and dashboard ?

**Comment by larrydevops** [comment]
I have the same issue, Saleor Api running in Docker container, dashboard and storefront running on local Docker host in dev mode.

![Screenshot 2020-05-30 at 16 45 26](https://user-images.githubusercontent.com/54898821/83331231-15ee7f80-a295-11ea-8277-e247f95b8521.png)

**System Information**
Debian 10

**Comment by Joash-star** [comment]
Any update

**Comment by JannikZed** [comment]
I solved the issue by setting the env variable `AWS_QUERYSTRING_AUTH: "True"` 
Like this, all generated S3 links for images have the correct token to access it. It's not documented anywhere.

**Comment by NyanKiyoshi** [comment]
@JannikZed if you are using for production, you should use cloudfront rather than S3 directly, this would solve your issue.

**Comment by JannikZed** [comment]
we are using minio, so no cloudfront available unfortunately. Maybe that's also why we came across this issue.

**Comment by sycts88** [comment]
Hi JannikZed, may i know are you using saleor 2.10 and above?
did you copy saleor media folder to AWS-S3?  I cannot seem to find media folder, what i saw in saleor master is static folder

**Comment by sycts88** [comment]
@NyanKiyoshi cloudfront is also need to copy media file over?

**Comment by sycts88** [comment]
Does saleor master 2.10 above has media folder...because when i cloned it i cannot see there is media folder instead there is only static folder

**Comment by sycts88** [comment]
I have added AWS_QUERYSTRING_AUTH: "True" in heroku config but still no images displayed

**Comment by JannikZed** [comment]
hey @sycts88 . I deployed saleor using our helm chart: https://github.com/trieb-work/Saleor-Helm-Chart.
I set the following values:
```
  AWS_S3_ENDPOINT_URL:
  AWS_S3_CUSTOM_DOMAIN:
  AWS_ACCESS_KEY_ID:
  AWS_STORAGE_BUCKET_NAME:
  AWS_MEDIA_BUCKET_NAME:
  AWS_MEDIA_CUSTOM_DOMAIN:
  AWS_QUERYSTRING_AUTH: True
```
I don't have a media folder. I called the bucket `static`, but you can name it whatever you want. 
I have no idea about cloudfront, but I don't think that you need to copy anything over.

**Comment by sycts88** [comment]
Thank you @JannikZed i have checked my heroku saleor master i don't have AWS_STORAGE_BUCKET_NAME (what is it for ?) because i thought i only have one media name in AWS.
For AWS_S3_ENDPOINT_URL is it saleor master URL like ''.herokuapp.com"

Yup, me too i don't find any media folder in saleor master 2.10.x.  i only saw saleor->static->images and saleor -> static->placeholders

**Comment by sycts88** [comment]
Hi @JannikZed i kind of figure out on the AWS_STORAGE_BUCKET_NAME to asitis1 (my AWS bucket name)  and AWS_MEDIA_BUCKET_NAME: static.  After i changed this two settings on my saleor master heroku setting. I  turn-on my saleor dashboard heroku, and goto catalog --> Products but still no images.

**Comment by sycts88** [comment]
![image](https://user-images.githubusercontent.com/59194717/84668575-641ea680-af56-11ea-8603-e2a5afb8f578.png)

**Comment by sycts88** [comment]
In Saleor master heroku settings, I don't have AWS_S3_ENDPOINT_URL but i have set allowed_host as .herokuapp.com.

**Comment by sycts88** [comment]
![image](https://user-images.githubusercontent.com/59194717/84669207-2ec68880-af57-11ea-8d32-f165a13c239f.png)
There is no images in static-->images.

But saleor master static --> placeholders has images.
![image](https://user-images.githubusercontent.com/59194717/84669400-66cdcb80-af57-11ea-840d-10bd7d539582.png)

**Comment by sycts88** [comment]
I have run my saleor dashboard and it says DevTools failed to load SourceMap: Could not load content for chrome-extension://fheoggkfdfchfphceeifdbepaooicaho/sourceMap/chrome/iframe_handler.map: HTTP error: status code 404,  net::ERR_UNKNOWN_URL_SCHEME

**Comment by sycts88** [comment]
![image](https://user-images.githubusercontent.com/59194717/84908132-a7a81a80-b0e6-11ea-8758-de768e629dcb.png)

**Comment by Rohoyangu** [comment]
@sycts88  Did you solve?

**Comment by sycts88** [comment]
@Rohoyangu 
no did not solve it...when i view the images in S3 i can see images.  When i m in saleor dashboard or storefront still not able to see any images

**Comment by sycts88** [comment]
Hi, i hv removed S3 and just using saleor dashboard images existing folder. But still not able to see the images in saleor

**Comment by ghost** [comment]
Please fix this bug.

**Comment by 2mwayungi** [comment]
@sycts88  I solved by changing version of saleor to 2.9

**Comment by Olegt0rr** [comment]
The same problem:
404 at `{api_host}/media/__sized__/products/saleordemoproduct_paints_01-thumbnail-510x510.png`

**Comment by mehrdad-shokri** [comment]
I'm having this exact problem now. I'm using minio and AWS envs as @JannikZed has suggested. ``collectstatic`` works fine, even links are generated correctly on dashboard, then only problem? no token is added to link. so it's something like this: ``https://static.example.com/__sized__/products/saleordemoproduct_cl_bogo03_1-thumbnail-255x255.png``. any ideas, please?  
I'm on saleor v3 + minio/minio:RELEASE.2021-06-17T00-10-46Z  
I can confirm that saleor has connected to minio correctly since both static and media buckets are populated.

**Reply by krzysztofwolski** [reply]
Communication with a bucket is working, so connection issues are sorted out.
2 questions:
- Are thumbnails generated? (it should be done automatically with worker instance)
- Could it be a permissions issue, where the bucket is not configured to host images publicly?

**Reply by gleniat** [reply]
Try `docker-compose run --rm api python manage.py create_thumbnails`.
This probably won't fix worker not generating newly uploaded images.

**Comment by mehrdad-shokri** [comment]
I had multiple problems but finally solved them out. First set bucket policy to download. second fix ``AWS_STATIC_CUSTOM_DOMAIN`` ``AWS_MEDIA_CUSTOM_DOMAIN`` values to point to bucket url.  
Right now I'm stuck on building/codegen for storefront. I'm using master branch for both storefront and api. Api is working correctly and I can confirm that by browsing dashboard and doing stuff.  But storefront gives error upon running ``npm run build/codegen/codegen-sdk``  
@krzysztofwolski can you confirm this is a bug due to the fact that ``AttributeFilterInput`` has not ``channel`` param on it?  
Also this brings the question of multi-channel readiness. I once read v3 will be released around Jul/Aug, with this param missing on ``AttributeFilterInput`` so every product/cattegory/collection can't be reterieved by channel. Seems like hasn't been implemented on backend side completely.  
For reference please check these issues:  

- https://github.com/mirumee/saleor-storefront/issues/1067
- https://github.com/mirumee/saleor-storefront/issues/1071
- https://github.com/mirumee/saleor/issues/7601
